src/Orleans.Runtime/Versions/GrainVersionStore.cs

```diff
@@ -14,21 +17,13 @@ internal class GrainVersionStore : IVersionStore
         private readonly string clusterId;
         private IVersionStoreGrain StoreGrain => this.grainFactory.GetGrain<IVersionStoreGrain>(this.clusterId);
 
-        public bool IsEnabled { get; private set; }
+        public bool IsEnabled { get; }
 
-        public GrainVersionStore(IInternalGrainFactory grainFactory, ILocalSiloDetails siloDetails)
+        public GrainVersionStore(IInternalGrainFactory grainFactory, ILocalSiloDetails siloDetails, IServiceProvider services)
         {
             this.grainFactory = grainFactory;
             this.clusterId = siloDetails.ClusterId;
-            this.IsEnabled = false;
-        }
-
-        public void SetStorageManager(IStorageProviderManager storageProviderManager)
-        {
-            IStorageProvider unused;
-            IsEnabled = storageProviderManager.TryGetProvider(
-                ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME,
-                out unused);
+            this.IsEnabled = services.GetServices<IKeyedService<string,IStorageProvider>>().Any(s => s.Equals(ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME));
```

Comment:
I think we can use `services.GetService<IStorageProvider>` here to see if default storage provider is configured

Reply:
those calls were unnecessarily asynchrony, but don't think that matters really.  Will use them, it's better than code duplication.